salt/pillar/http_yaml.py

```diff
@@ -97,7 +99,12 @@ def ext_pillar(minion_id,
             url = re.sub('<{0}>'.format(grain_name), grain_value, url)
 
     log.debug('Getting url: %s', url)
-    data = __salt__['http.query'](url=url, decode=True, decode_type='yaml')
+
+    if 'username' in kwargs and 'password' in kwargs:
+        data = __salt__['http.query'](url=url, decode=True, decode_type='yaml',
+                                      username=kwargs['username'], password=kwargs['password'])
+    else:
+        data = __salt__['http.query'](url=url, decode=True, decode_type='yaml')
```

Comment:
And if we just specify the defaults as none, then they can just be passed through as none anyway, and you don't need this if statement.

`data = __salt__['http.query'](url=url, decode=True, decode_type='yaml', username=username, password=password)`

Reply:
Thanks that would be way more elegant. I will try to get those changes in the pr asap.